pkg/migrate/migrate.go

```diff
@@ -660,24 +707,30 @@ func mutateSC(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
 // it will also cleanup WIP migration pods it discovers that happen to be mounting a supplied PVC.
 // if a pod is not created by pvmigrate, and is not controlled by a statefulset/deployment, this function will return an error.
 // if waitForCleanup is true, after scaling down deployments/statefulsets it will wait for all pods to be deleted.
-func scaleDownPods(ctx context.Context, w *log.Logger, clientset k8sclient.Interface, matchingPVCs map[string][]corev1.PersistentVolumeClaim, checkInterval time.Duration) error {
+// NOTE: this funcation updates the matchingPVCs map with the pod which consumes the pvc
```

Comment:
they not have the getPVCs function return the pvcCtx with the pod rather than mutating the argument?

Reply:
yeah I do not like mutating arguments
that seems like it'll bite us

Reply:
I refactored this to not have side effects.